lib/scout_apm/auto_instrument/layer.rb

```diff
@@ -0,0 +1,21 @@
+
+module ScoutApm
+  def self.AutoInstrument(name)
+    request = ScoutApm::RequestManager.lookup
+
+    begin
+      layer = ScoutApm::Layer.new('AutoInstrument', name)
+      request.start_layer(layer)
+      started_layer = true
+
+      result = yield
+    rescue
+      request.error!
```

Comment:
`error!` should only be marked true if the controller lets an exception escape, but this will mark a request as an error if any individual call exceptions, even if caught right away.

If the controller layer still exists, we should let it handle this error! mark.

During the quick test on APM prod, some code like below was being marked as an "error" due to this.

```
def index
  something_that_raises
rescue
  render text: "ok"
end
```

Reply:
That makes sense. So perhaps we shouldn't handle errors and just propagate them up?

Reply:
Okay done.